packages/controls/src/serialization/function-serialization.ts

```diff
@@ -44,30 +53,46 @@ export function serializeFunction<T extends AnyFunction>(
   return messageChannel.port2 as SerializedFunction<T>
 }
 
-export function deserializeFunction<T extends AnyFunction>(
-  serializedFunction: SerializedFunction<T>,
-): DeserializedFunction<T> {
-  type Result = SerializedFunctionReturnType<T>
-  type ResultMessageEvent = MessageEvent<[CallID, Result]>
-  type ResolveCallPromise = (value: Result | PromiseLike<Result>) => void
-
+function onmessageHandler<T extends AnyFunction>(): OnMessageHandler<T> {
+  type ResultMessageEvent = MessageEvent<
+    [CallID, SerializedFunctionReturnType<T>]
+  >
   let nextCallId = 0
-  const calls = new Map<CallID, ResolveCallPromise>()
+  const calls = new Map<CallID, ResolveCallPromise<T>>()
 
-  serializedFunction.onmessage = ({
```

Comment:
When `deserializeFunction` is called twice on the same `serializedFunction` instance, reassigning `serializedFunction.onmessage` here overrides/orphans any previous message handlers. That means:

- function calls to previously deserialized functions will never return, since the corresponding promises are never resolved
- these unresolved `Promises` effectively constitute a memory leak